src/clean-core/capped_array.hh

```diff
@@ -55,14 +55,13 @@ public:
     {
         CC_CONTRACT(size <= N);
 
-        if constexpr (!std::is_trivially_constructible_v<T>)
-        {
-            // NOTE: DO NOT use array-placement new!
-            // it stores the array size in the first 8 bytes
-            // (at least on MSVC, the standard simply allows a padding on the return value)
-            for (compact_size_t i = 0; i < _size; ++i)
-                new (placement_new, &_u.value[i]) T();
-        }
+        // NOTE: DO NOT use array-placement new!
+        //       it stores the array size in the first 8 bytes
+        //       (at least on MSVC, the standard simply allows a padding on the return value)
+        // NOTE: default constructor default-initializes all members
+        //       if the overhead is too big, use capped_array::uninitialized
```

Comment:
This comment should explain the intricacies for primitive types i think. If you come across this code and ask yourself why this isn't conditioned on non-trivial ctors, these lines don't make it clear

Reply:
added a line. Though this is a deprecated datatype, so too much description is definitely wasted. This is a good point for the eventual overall container summary though.